app/api/attendees.py

```diff
@@ -31,11 +35,26 @@ def before_post(self, args, kwargs, data):
         """
         require_relationship(['ticket', 'event'], data)
 
-        ticket = db.session.query(Ticket).filter_by(id=int(data['ticket'])).first()
+        ticket = db.session.query(Ticket).filter_by(
+            id=int(data['ticket'])
+            ).first()
         if ticket is None:
-            raise UnprocessableEntity({'pointer': '/data/relationships/ticket'}, "Invalid Ticket")
+            raise UnprocessableEntity(
+                {'pointer': '/data/relationships/ticket'}, "Invalid Ticket"
+                )
         if ticket.event_id != int(data['event']):
-            raise UnprocessableEntity({'pointer': '/data/relationships/ticket'}, "Ticket belongs to a different Event")
+            raise UnprocessableEntity(
+                {'pointer': '/data/relationships/ticket'},
+                "Ticket belongs to a different Event"
+            )
+
+        if db.session.query(TicketHolder.id).filter_by(
+                ticket_id=int(data['ticket']), event_id=int(data['event'])
```

Comment:
Checking only for the `ticket_id` should be sufficient. If the `ticket_id` already exists, then that already means that we have someone on this ticket. 
There won't be a case when the `ticket_id` is same but `event_id` is different. Hence only filtering by `ticket_id` is sufficient IMO. 

Thoughts @bhaveshAn @Kreijstal @mayank8318

Reply:
I was suspecting that. Let's see what others have to say in this.

Reply:
I agree with @srv-twry

Reply:
I don't think it matters much, since it will pretty much bring the same result (unless something extraordinary happens), but if it improves redability to just filter by ticket, then be my guest.